### PR TITLE
[refactor](nereids)make NormalizeAggregate rule more clear and readable

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeAggregate.java [hunk 1]
```diff
@@ -100,22 +99,74 @@ public class NormalizeAggregate extends OneRewriteRuleFactory implements Normali
...
+            // The LogicalAggregate node may contain window agg functions and usual agg functions
+            // we call window agg functions as window-agg and usual agg functions as trival-agg for short
+            // This rule simplify LogicalAggregate node by:
+            // 1. Push down some exprs from old LogicalAggregate node to a new child LogicalProject Node,
+            // 2. create a new LogicalAggregate with normalized group by exprs and trival-aggs
+            // 3. Pull up normalized old LogicalAggregate's output exprs to a new parent LogicalProject Node
+            // Push down exprs:
+            // 1. all group by exprs
+            // 2. subquery expr in trival-agg
+            // 3. all input slots of trival-agg
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeAggregate.java:111]: should include WindowExpression inner trival-agg

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeAggregate.java [hunk 2]
```diff
@@ -100,22 +101,94 @@ public class NormalizeAggregate extends OneRewriteRuleFactory implements Normali
...
+            // TRUE part 1: need push down itself, if it contains subqury or window expression
+            // FALSE part 2: need push down its input slots, if it DOES NOT contain subqury or window expression
+            Map<Boolean, Set<Expression>> categorizedNoDistinctAggsChildren = aggFuncs.stream()
+                    .filter(aggFunc -> !aggFunc.isDistinct())
+                    .flatMap(agg -> agg.children().stream())
+                    .collect(Collectors.groupingBy(
+                            child -> child.containsType(SubqueryExpr.class, WindowExpression.class),
+                            Collectors.toSet()));
+
+            // split non-distinct agg child as two parts
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeAggregate.java:148]:
```suggestion
            // split distinct agg child as two parts
```

**Review by morrySnow — DISMISSED** [review]

**Comment by starocean999** [comment]
run buildall

**Review by xzj7019 — APPROVED** [review]

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**